src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -315,9 +337,24 @@ impl InflightActorState {
         assert_eq!(prev_epoch, epoch.prev);
         let (min_mutation_epoch, _) = self.barrier_mutations.pop_first().expect("should exist");
         assert_eq!(min_mutation_epoch, epoch.prev);
+        match &self.status {
+            InflightActorStatus::NotStarted => {
+                unreachable!("should have issued a barrier when collect")
+            }
+            InflightActorStatus::IssuedFirst(pending_barriers) => {
+                assert_eq!(
+                    prev_epoch,
+                    pending_barriers.first().expect("non-empty").epoch.prev
+                );
+                self.status = InflightActorStatus::Running(
```

Comment:
Does this mean that, by `collect`ing, we are sure that all barrier senders in this actor have been registered?

Reply:
Yes, this is expected, because actors should be able to collect the barrier only after the barrier is sent, and therefore when a barrier is collected, all barrier senders should have been registered.